<?xml version="1.0"?>

<!--Edited from hybrid_astar/manual.launch-->

<launch>

  <arg name="geotiff_map_file_path" default="$(find wheego)/maps"/>

  <param name="/use_sim_time" value="true"/>

  <node pkg="tf" type="static_transform_publisher"
    name="base_to_laser_broadcaster"
    args="0 0 0 0 0 0 base_footprint velodyne 100"/>

  <include file="$(find wheego)/launch/mapping_edit.launch">
    <arg name="map_size" value="1200"/>
    <arg name="resolution" value="0.25"/>
    <arg name="map_start_x" value="-0.00333333333"/> <!-- -1/map_size-->
    <arg name="map_start_y" value="-0.00333333333"/>
    <arg name="odom_frame" value="base_footprint"/>
    <arg name="scan_topic" value="scan2"/>
  </include>

  <include file="$(find wheego)/launch/cloud_to_scan.launch"/>

  <node name="map_handler" pkg="wheego" type="map_handler.py"/>

  <include file="$(find hector_geotiff)/launch/geotiff_mapper.launch">
    <arg name="trajectory_source_frame_name" value="scanmatcher_frame"/>
    <arg name="map_file_path" value="$(arg geotiff_map_file_path)"/>
  </include>

  <node name="hybrid_astar" pkg="hybrid_astar" type="hybrid_astar">
    <remap from="initialpose" to="pose_nav"/>
    <remap from="map" to="map_nav"/>
  </node>
  
  <node name="map_server" pkg="map_server" type="map_server"
    args="$(find wheego)/maps/outLarge.yaml">
    <remap from="map" to="map_nav"/>
  </node>
  <node pkg="tf" type="static_transform_publisher"
    name="odom_map_nav_bcast"
    args="0 0 0 0 0 0 odom map_nav 100"/>
  <node pkg="tf" type="static_transform_publisher"
    name="map_nav_path_bcast"
    args="0 0 0 0 0 0 map_nav path 100"/>
  <node pkg="tf" type="static_transform_publisher"
    name="map_nav_map_bcast"
    args="0 0 0 0 0 0 map_nav map 100"/>
  <node name="rviz" pkg="rviz" type="rviz"
    args="-d $(find wheego)/launch/mapNav.rviz" />
</launch>
